common/tools/dev-tool/src/util/customization/interfaces.ts

```diff
@@ -20,6 +26,18 @@ export function augmentInterface(
   originalInterface: InterfaceDeclaration | undefined,
   originalFile: SourceFile
 ) {
+  const annotation = getAnnotation(customInterface);
+  /* If the interface is marked with `// @azsdk-remove`, we'll remove it from the original file */
+  if (annotation?.type === "remove") {
+    originalInterface?.remove();
+    return;
+  }
+
+  /* If the interface is marked with `// @azsdk-rename`, we'll rename the original interface */
+  if (annotation?.type === "rename" && annotation.param) {
+    originalInterface?.rename(customInterface.getName());
```

Comment:
Could you add a test to check that every reference to a renamed interface is also renamed?

Reply:
sure, that's handled by ts-morph though

Reply:
added :)